File: app/src/main/java/com/example/moneygement/controller/InputExpenseHouseholdAccountBookActivity.java
package com.example.moneygement.controller;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import com.example.moneygement.R;

public class InputExpenseHouseholdAccountBookActivity extends AppCompatActivity{

    private String expenseYear;
    private String expenseMonth;
    private String expenseDay;
    private String expenseDate;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_input_household_account_book);

        TextView date = (TextView)findViewById(R.id.date);
        //Intentのオブジェクトを生成する
        Intent rIntent  = getIntent();
        //年月日を受け取る
        expenseYear = String.valueOf(rIntent.getIntExtra("expenseYear",0));
        expenseMonth = String.valueOf(rIntent.getIntExtra("expenseMonth",0));
        expenseDay = String.valueOf(rIntent.getIntExtra("expenseDay",0));

        expenseDate = expenseYear+"年"+expenseMonth +"月"+expenseDay+ "日";

        date.setText(expenseDate);

    }

    @Override
    protected void onResume(){
        super.onResume();

        Button continueButton = (Button)findViewById(R.id.tocontinue);
        Button saveBtn = (Button)findViewById(R.id.savebutton);
        Button cancelBtn = (Button)findViewById(R.id.cancel);

        continueButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //エディットテキストを探す
                EditText Amountofmoney = (EditText) findViewById(R.id.editTextNumberSigned);
                EditText textPersonName = (EditText) findViewById(R.id.editTextTextPersonName);
                Spinner textcategory = (Spinner) findViewById(R.id.spinnerCategory);

                String txtmoney = Amountofmoney.getText().toString();
                String personName = textPersonName.getText().toString();
                String category = textcategory.getSelectedItem().toString();

//              保存処理

                Intent intent = new Intent(InputExpenseHouseholdAccountBookActivity.this, InputExpenseHouseholdAccountBookActivity.class);
                int money = Integer.parseInt("txtmoney");
                intent.putExtra("money", money);
                intent.putExtra("personName", personName);
                intent.putExtra("cotegory", category);
                startActivity(intent);
            }
        });

        saveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText amounToMoney = (EditText)findViewById(R.id.editTextNumberSigned);
                EditText textPersonName = (EditText)findViewById(R.id.editTextTextPersonName);
                Spinner textCategory = (Spinner) findViewById(R.id.spinnerCategory);

                String txtmoney = amounToMoney.getText().toString();
                String personName = textPersonName.getText().toString();
                String category = textCategory.getSelectedItem().toString();

//              保存処理

                Intent intent = new Intent(InputExpenseHouseholdAccountBookActivity.this, DispLedgerActivity.class);
                startActivity(intent);
            }
        });

        cancelBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(InputExpenseHouseholdAccountBookActivity.this, DispCalenderExpensesLedgerActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/moneygement/controller/InputExpenseHouseholdAccountBookActivity.kt
package com.example.moneygement.controller

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.CreateExpenseDetailMutation
import com.example.LedgerQuery
import com.example.LedgersQuery
import com.example.moneygement.R
import com.example.moneygement.controller.InputExpenseHouseholdAccountBookActivity
import com.example.moneygement.repository.Ledger
import kotlinx.android.synthetic.main.activity_input_share_income.*
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class InputExpenseHouseholdAccountBookActivity : AppCompatActivity() {
    private lateinit var expenseYear: String
    private lateinit var expenseMonth: String
    private lateinit var expenseDay: String
    private lateinit var expenseDate: String
    private lateinit var insertDate: String
    private var ledgerId: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_input_household_account_book)
        val dateTextView = findViewById<View>(R.id.date) as TextView
        //Intentのオブジェクトを生成する
        val rIntent = intent

        ledgerId = rIntent.getIntExtra("ledgerId", 0)
        //年月日を受け取る
        expenseDate = rIntent.getStringExtra("checkDay")
        dateTextView.text = expenseDate
        insertDate = expenseDate.replace("[年月]".toRegex(), "-")
        insertDate = insertDate.replace("日","")
    }

    override fun onResume() {
        super.onResume()

        val continueButton = findViewById<View>(R.id.tocontinue) as Button
        val saveBtn = findViewById<View>(R.id.savebutton) as Button
        val cancelBtn = findViewById<View>(R.id.cancel) as Button
        val amounToMoney = findViewById<View>(R.id.editTextNumberSigned) as EditText
        val textPersonName = findViewById<View>(R.id.editTextTextPersonName) as EditText
        val textCategory = findViewById<View>(R.id.spinner) as Spinner

        continueButton.setOnClickListener { //エディットテキストを探す
            val money = amounToMoney.text.toString().toInt()
            val note = textPersonName.text.toString()
            val categoryId = textCategory.selectedItemPosition


//              保存処理
            val createExpenseDetailMutation = CreateExpenseDetailMutation.builder()
                    .ledgerId(ledgerId)
                    .categoryId(categoryId + 1)
                    .date(insertDate)
                    .amount(money)
                    .note(note)
                    .build()

            Ledger().insertExpense(createExpenseDetailMutation)

            val intent = Intent(this@InputExpenseHouseholdAccountBookActivity, InputExpenseHouseholdAccountBookActivity::class.java)
            intent.putExtra("checkDay", expenseDate)
            intent.putExtra("ledgerId", ledgerId)
            startActivity(intent)
        }
        saveBtn.setOnClickListener {
            val money = amounToMoney.text.toString().toInt()
            val note = textPersonName.text.toString()
            val categoryId = textCategory.selectedItemPosition


            //              保存処理
            val createExpenseDetailMutation = CreateExpenseDetailMutation.builder()
                    .ledgerId(ledgerId)
                    .categoryId(categoryId + 1)
                    .date(insertDate)
                    .amount(money)
                    .note(note)
                    .build()

            Ledger().insertExpense(createExpenseDetailMutation)

            val intent = Intent(this@InputExpenseHouseholdAccountBookActivity, DispLedgerActivity::class.java)
            startActivity(intent)
        }
        cancelBtn.setOnClickListener {
            val intent = Intent(this@InputExpenseHouseholdAccountBookActivity, DispCalenderExpensesLedgerActivity::class.java)
            intent.putExtra("ledgerId", ledgerId)
            startActivity(intent)
        }
    }
}